samples/linux/subscribe_publish_sample/subscribe_publish_sample.c

```diff
@@ -78,13 +78,16 @@ static void disconnectCallbackHandler(AWS_IoT_Client *pClient, void *data) {
 
 	if(aws_iot_is_autoreconnect_enabled(pClient)) {
 		IOT_INFO("Auto Reconnect is enabled, Reconnecting attempt will start now");
+		pClient->clientStatus.clientState = CLIENT_STATE_PENDING_RECONNECT;
```

Comment:
You should not make any changes to the state in this file.  State changes will be handled by API functions.

Reply:
ok, I will remove those state changes code.